v1 and v2 specs (#276)
zarr-developers/zarr-specs: docs/v1/v1.0.rst

    array. 'C' means row-major order, i.e., the last dimension varies fastest;
    'F' means column-major order, i.e., the first dimension varies fastest.

Other keys MAY be present within the metadata object however they MUST
NOT alter the interpretation of the required fields defined above.

For example, the JSON object below defines a 2-dimensional array of
64-bit little-endian floating point numbers with 10000 rows and 10000
columns, divided into chunks of 1000 rows and 1000 columns (so there
will be 100 chunks in total arranged in a 10 by 10 grid). Within each
chunk the data are laid out in C contiguous order, and each chunk is
compressed using the Blosc compression library::

    {
        "chunks": [
            1000,
            1000
        ],
        "compression": "blosc",
        "compression_opts": {
            "clevel": 5,
            "cname": "lz4",
            "shuffle": 1
        },
        "dtype": "<f8",
        "fill_value": null,
        "order": "C",
        "shape": [
            10000,
            10000
        ],
        "zarr_format": 1
    }

Data type encoding
~~~~~~~~~~~~~~~~~~

Simple data types are encoded within the array metadata resource as a

> string, following the `NumPy array protocol type string (typestr)
> format
> <numpy:arrays.interface>`_. The

Reviewer: ```suggestion
string, following the `NumPy array protocol type string (typestr)
format
<https://numpy.org/doc/stable/reference/arrays.interface.html>`_. The
```

This was giving 404. I've updated the link.

format consists of 3 parts: a character describing the byteorder of
the data (``<``: little-endian, ``>``: big-endian, ``|``:
not-relevant), a character code giving the basic type of the array,
and an integer providing the number of bytes the type uses. The byte
order MUST be specified. E.g., ``"<f8"``, ``">i4"``, ``"|b1"`` and
``"|S12"`` are valid data types.

Structure data types (i.e., with multiple named fields) are encoded as

> a list of two-element lists, following `NumPy array protocol type
> descriptions (descr)
> <numpy:arrays.interface>`_.

Reviewer: ```suggestion
a list of two-element lists, following `NumPy array protocol type
descriptions (descr)
<https://numpy.org/doc/stable/reference/arrays.interface.html>`_.
```

This was also giving 404. Updated the link.

For example, the JSON list ``[["r", "|u1"], ["g", "|u1"], ["b",
"|u1"]]`` defines a data type composed of three single-byte unsigned
integers labelled 'r', 'g' and 'b'.

Chunks
------

Each chunk of the array is compressed by passing the raw bytes for the
chunk through the primary compression library to obtain a new sequence
of bytes comprising the compressed chunk data. No header is added to
the compressed bytes or any other modification made. The internal
structure of the compressed bytes will depend on which primary

> compressor was used. For example, the `Blosc compressor
> <https://github.com/Blosc/c-blosc/blob/main/README_HEADER.rst>`_

Reviewer: ```suggestion
compressor was used. For example, the `Blosc compressor
<https://github.com/Blosc/c-blosc/blob/main/README_CHUNK_FORMAT.rst>`_
```
The `README_HEADER.rst` has been renamed to `README_CHUNK_FORMAT.rst`.
